doc/vxlan/EVPN/EVPN_VxLAN_Multihoming.md

```diff
@@ -0,0 +1,1450 @@
+1. Handling of L2-NHG netlink notifications and dump from kernel.
+2. Programming of L2_NEXTHOP_GROUP_TABLE APP-DB entries based on kernel L2-NHG events.
+3. Programming of VXLAN_FDB_TABLE APP-DB entries with L2-NHID.
+4. Warm-reboot reconciliation of L2_NEXTHOP_GROUP_TABLE and VXLAN_FDB_TABLE similar to ROUTE_TABLE/NEXTHOP_GROUP_TABLE reconciliation in Fpmsyncd.
+
+### 3.3.6 Vxlanorch changes
+- Changes required to maintain refcnt to a dip per L2 NHG.
+
+### 3.3.7 Fdborch changes
+- FdbOrch is enhanced to process EVPN MH remote MAC updates via existing VXLAN_FDB_TABLE.
+- FdbOrch is extended to process new field "ifname" from VXLAN_FDB_TABLE entries.
+- FdbOrch is extended to process new field "type" for possible local/remote from VXLAN_FDB_TABLE entries.
+- FdbOrch is store and update MAC entries per ES associated PortChannel. This is to update the MAC entries when the associated PortChannel interface is going up/down. 
+- FdbOrch is interact with EvpnMhOrch to check if the given PortChanenl is associated with ES or not, fetch the L2-NHID associated with PortChannel, etc.
+- FdbOrch is extended to process FDB entries on local po down and program then pointing to backup nexthop group using EVPN_ES_BACKUP_NHG_TABLE.
+
+Fdborch to process the local po down and up in a similar fashion as for MCLAG. 
+
+### 3.3.8 L2nhgorch changes
+- New Orchagent class being introduced.
+- Subscription to different tables
+  - APP DB - L2_NEXTHOP_GROUP, L2_NEXTHOP_GROUP_MEMBER
+  - CONFIG DB - EVPN_ES_INTERFACE.
+- Subscription to L2_NEXTHOP_GROUP
+  - Creates/deletes the L2_ECMP_GROUP ASIC DB table. 
+  - Interacts with PortsOrch to create a bridge-port object of type L2_ECMP_GROUP.
+- Subscription to L2_NEXTHOP_GROUP_MEMBER
+  - Creates/deletes the L2_ECMP_GROUP_MEMBER ASIC DB tables entries on the L2_NEXTHOP_GROUP APP DB table updates.
+  - Interacts with VxlanOrch to update the refcnt per remote VTEP.
+- Subscription to EVPN_ESI CONFIG_DB table 
+  - Informs Fdborch of the ESI to PortChannel mapping.
+  - Maintain a local cache of the mapping.
+- Subscription to LAG oper-status updates from PortsOrch
+- API 
+  - L2 NHID to Bridgeport mapping.
+
+### 3.3.9 EvpnMhOrch
+A new EvpnMhOrch class will be introduced to perform following activities, 
+
+1. Subscribe to EVPN_MH_GLOBAL config table and set the SAI switch attribute. 
+2. Subscribe to EVPN_ES_INTERFACE config table updates and maintain internal cache.
+3. Provide set of API to check if the given PortChannel is ES associated or not. 
+4. Subscribe to LAG_TABLE oper status changes and trigger MAC update requests to FdbOrch.
+
+## 3.4 SAI
+
+### 3.4.1 New SAI Objects 
+
+- **L2_ECMP_GROUP**
+  - Holds a set of remote VTEPs.
+  - Bridgeport objects of type L2_ECMP_GROUP will be associated with this object. 
+  - Known unicast traffic with Destination MACs pointing to the above bridgeport type will be load balanced among the remote VTEPs. 
+
+```
+typedef enum _sai_l2_ecmp_group_attr_t
```

Comment:
it is noted that l2 ecmp group has almost the same attribute with sainexthopgroup.
 why not  to reuse sainexthopgroup object and add some attribute to identify l2 and l3 ecmp group?

Reply:
@philo-micas , yes, it makes sense. SAI subsection is updated.